### DISCUSSION TITLE
Process hint brushes first

### DISCUSSION THREAD

**Discussion by hemebond** [ericwa/ericw-tools#348] [Ideas]
I was wondering what would happen to the brush and vis splitting (those the terms?) if hint brushes were processed first, as right now they seem to be processed last.

I originally hoped hint brushes could create "rooms" that would be split later by the geometry, reducing the chance of an odd split affecting far away places in the map.

(apologies if this isn't how it actually works, I can only go by how it _appears_ to behave when I use it)

**Comment by Paril** [comment]
Hint brushes basically just force a BSP split to occur on their face by being extremely high priority. They're used to increase density of portals in areas that are being under-represented; for instance if you have a very long pipe, you can use a hint brush to cut the inner tube into two, otherwise the entire tube might just be two portals (one at the start & one at the end) and would flow visibility through the whole thing (since visibility is determined through flowing portals, you need three portals to effectively cut off visibility to another area - the source portal, the portals between the destination, and then the destination itself).
